cli/src/main.rs

```diff
@@ -580,53 +580,118 @@ fn run<I: IntoIterator<Item = T>, T: Into<OsString> + Clone>(args: I) -> Result<
-                            .takes_value(true)
-                            .multiple(true)
-                            .help("Metadata to include with node (<key>=<value>)"),
-                    )
-                    .arg(
-                        Arg::with_name("private_key_file")
-                            .value_name("private-key-file")
-                            .short("k")
-                            .long("key")
-                            .takes_value(true)
-                            .help("Name or path of private key"),
-                    ),
+                ))
+                .arg(
+                    Arg::with_name("force")
+                        .long("force")
+                        .help("Overwrite node if it already exists"),
+                )
+                .arg(
+                    Arg::with_name("status_url")
+                        .takes_value(true)
+                        .help("URL of splinter REST API to query for node data"),
+                )
+                .arg(
+                    Arg::with_name("key_files")
+                        .long("key-file")
+                        .takes_value(true)
+                        .multiple(true)
+                        .required(true)
+                        .help("Path of public key file to include with node"),
+                )
+                .arg(
+                    Arg::with_name("metadata")
+                        .long("metadata")
+                        .takes_value(true)
+                        .multiple(true)
+                        .help("Metadata to include with node (<key>=<value>)"),
+                )
+                .arg(
+                    Arg::with_name("private_key_file")
+                        .value_name("private-key-file")
+                        .short("k")
+                        .long("key")
+                        .takes_value(true)
+                        .help("Name or path of private key"),
+                ),
+        );
+
+    #[cfg(feature = "registry-add-node")]
+    let registry_command = registry_command.subcommand(
+        SubCommand::with_name("add")
+            .about("Add a node to the local registry")
+            .arg(
+                Arg::with_name("display_name")
+                    .long("display")
+                    .takes_value(true)
+                    .help("Human-readable name for the new node"),
+            )
+            .arg(
+                Arg::with_name("dry_run")
+                    .long("dry-run")
+                    .help("Show the expected changes without submitting the node"),
+            )
+            .arg(
+                Arg::with_name("endpoint")
+                    .long("endpoint")
+                    .takes_value(true)
+                    .multiple(true)
+                    .required_unless("from_remote")
+                    .help("Network endpoint for the new node"),
+            )
+            .arg(
+                Arg::with_name("from_remote")
```

Comment:
I actually think that all the options for modifying the node (display name, endpoints, keys, metadata) should not be allowed when using `--from-remote`. If you take a look at the design we have for the update command, the ways in which you can modify these things is pretty complicated; the add command doesn't support removing or replacing some of these values. So IMO we're better off disabling any modifications when copying the node from remote to local.

Reply:
if `--from-remote` is used now the node will be copied from the remote registry without changing any information about it

Reply:
To make this clear and explicit to the user, you should add `.conflicts_with_all(&["display_name", "endpoint", "identity", "key_files", "metadata"])`. This will actively prevent users from trying to change the node when using `--from-remote`. The man page should describe this behavior.

Reply:
Fixed, also added a man page for this subcommand that includes a note about this in the description